uitools/toolkitcpp/docs/Authenticator.md

```diff
@@ -0,0 +1,51 @@
+[![ArcGIS Maps SDK for Qt](https://img.shields.io/badge/ArcGIS%20Maps%20SDK%20for%20Qt-0b5394)](https://developers.arcgis.com/qt/) [![ArcGIS Maps SDK for Qt toolkit](https://img.shields.io/badge/ArcGIS%20Maps%20SDK%20for%20Qt%20toolkit-ea4d13)](https://github.com/Esri/arcgis-maps-sdk-toolkit-qt) [![Qt Quick UI components](https://img.shields.io/badge/Qt%20Qt%20Quick%20UI%20components-ea4d13)](../../toolkitcpp/)
+
+# Authenticator - Qt Quick UI/QML Type
+
+The [Authenticator- Qt Quick UI/QML Type](https://developers.arcgis.com/qt/toolkit/api-reference/qml-esri-arcgisruntime-toolkit-authenticator.html) handles authentication challenges and automatically launches the appropriate UI for each type of authentication. The following security formats are supported.
+
+- ArcGIS Token 
+- HTTP Basic 
+- HTTP Digest 
+- ArcGIS OAuth
+- SAML
+- IWA (Kerberos & NTLM)
+- PKI
+
+![Authenticator image](../../docs/images/authenticator_qml.png)
+
+## How to use
+
+To get started, declare an Authenticator in your QML file. 
+
+```qml
+Authenticator {
+    anchors.centerIn: parent
+}
+```
+
+A full tutorial for setting up the Authenticator is available in the [developer guide](https://developers.arcgis.com/qt/security-and-authentication/tutorials/access-services-with-oauth-2-0/)
+
+## ArcGIS Authentication
+Portals configured to use ArcGIS Authentication will use token authentication by default. To use OAuth, you must pass in an instance of `OAuthUserConfiguration` with the Portal URL, client ID, and redirect URL to the `OAuthUserConfigurationManager`.
+
+```cpp
+DemoApp::DemoApp(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(new PortalItem(s_itemId, new Portal(true /*loginRequired*/, this)), this))
+{
+    const auto redirectUrl = QString{"qt-maps-sdk-app:/oauth2/callback"};
+    auto* config = new OAuthUserConfiguration(QUrl{"https://www.arcgis.com/"}, s_clientId, redirectUrl, this);
+    OAuthUserConfigurationManager::instance()->addOAuthUserConfiguration(config); 
```

Comment:
```suggestion
    OAuthUserConfigurationManager::instance()->addConfiguration(config); 
```

I just went with `addConfiguration` to avoid redundancy with the class name.